src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arrState = [];
+  let newState = { ...state };
+
+  actions.forEach((action) => {
+    if (action.type === 'clear') {
+      newState = {};
+      arrState.push({ ...newState });
```

Comment:
[CODE STYLE] - You are pushing the newState to the arrState array inside each condition. According to the [DONT REPEAT YOURSELF] principle, you should perform this action once after the switch statement, as this needs to be done in every case.